### PR TITLE
Added error logs in getUserRepositories from provider

### PR DESCRIPTION
**Pull request by avikalpg**
### Summary by CodeRabbit

- Bug Fix: Enhanced error handling in `getUserRepositories.ts` utility. Now, if no repositories are received from the GitHub API, a console warning will be generated for better debugging.
- Bug Fix: Improved logging for failures when attempting to get user repositories from providers. This will aid in identifying and resolving issues more efficiently.

### CODE REVIEW

File: utils/providerAPI/getUserRepositories.ts [hunk 1]
```diff
@@ -139,6 +142,7 @@ export const getUserRepositories = async (session: Session) => {
 	await Promise.allSettled(allUserReposPromises).then((results) => {
 		results.forEach((result) => {
 			if (result.status !== 'fulfilled') {
+				console.error(`[getUserRepositories] Failed to get repositories of the user (id: ${session.user.id}, name: ${session.user.name}) from one of the providers`, result.reason);
```
[the remark this answers is not in the document]
Comment (avikalpg) [reply utils/providerAPI/getUserRepositories.ts:145]: It only returns from the callback function provided to the `forEach` function. It does not return from the whole function.
This is expected behaviour.

File: utils/providerAPI/getUserRepositories.ts [hunk 2]
```diff
@@ -81,6 +81,9 @@ const getUserRepositoriesForGitHub = async (access_key: string, authId?: string)
 				console.error(`[getUserRepositories] Error occurred while getting user repositories from GitHub API (provider-assigned id: ${authId}). Endpoint: ${endPoint}`, err.message);
 				throw err;
 			})
+		if (repos.length === 0) {
+			console.warn(`[getUserRepositories] No repositories received from GitHub API (provider-assigned id: ${authId}). Endpoint: ${endPoint}`);
+		}
```
[the remark this answers is not in the document]
Comment (avikalpg) [reply utils/providerAPI/getUserRepositories.ts:86]: @tapishr How seriously should I take this comment? Is this actually a security/privacy problem?

[the remark this answers is not in the document]
Comment (tapishr) [reply utils/providerAPI/getUserRepositories.ts:86]: .debug ke peeche rakh diya karo usually aisi info

[the remark this answers is not in the document]
Comment (tapishr) [reply utils/providerAPI/getUserRepositories.ts:86]: Its fine in this case though

**Review by tapishr — APPROVED** [review]

**Pull request merged**